fraud/sync.go

```diff
@@ -0,0 +1,136 @@
+package fraud
+
+import (
+	"context"
+	"time"
+
+	"github.com/ipfs/go-datastore"
+	"github.com/libp2p/go-libp2p-core/event"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/peer"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+
+	"github.com/celestiaorg/go-libp2p-messenger/serde"
+
+	pb "github.com/celestiaorg/celestia-node/fraud/pb"
+)
+
+func (f *ProofService) syncFraudProofs(ctx context.Context) {
```

Comment:
Please add more comments on the inside of this function, holding the reader's hand a little bit. You can see a great example of this in Hlib's `get_shares.go`.